### PR TITLE
#1999 - Add vector parameter for Ball2

### LINKED ISSUE #1999: Add vector parameter for Ball2
See https://github.com/JuliaReach/LazySets.jl/issues/481.

### PR DESCRIPTION
**Pull request by mforets**
Closes #1999.

### CODE REVIEW

File: src/Sets/Ball2.jl [hunk 1]
```diff
@@ -259,12 +262,12 @@ Random sampling with uniform distribution in `B` is computed using Muller's meth
...
 """
-function sample(B::Ball2{N}, nsamples::Int=1;
+function sample(B::Ball2{N, VN}, nsamples::Int=1;
                 rng::AbstractRNG=GLOBAL_RNG,
-                seed::Union{Int, Nothing}=nothing) where {N<:AbstractFloat}
+                seed::Union{Int, Nothing}=nothing) where {N<:AbstractFloat, VN<:AbstractVector{N}}
     require(:Distributions; fun_name="sample")
     n = dim(B)
-    D = Vector{Vector{N}}(undef, nsamples) # preallocate output
+    D = Vector{VN}(undef, nsamples) # preallocate output
```
Comment (schillic) [inline src/Sets/Ball2.jl]: You cannot know that the samples will have the same vector type as the set.

Comment (mforets) [reply src/Sets/Ball2.jl]: That's weird.. but possible. So you suggest to hardcode to vector?

Comment (mforets) [reply src/Sets/Ball2.jl]: i checked and https://github.com/JuliaReach/LazySets.jl/blob/master/src/Utils/samples.jl#L187 is only defined for cache D with Vector. this means that `D = Vector{VN}(undef, nsamples)` will not work if `VN` is different from vector. ill open a separate issue, and revert my change here.

Comment (mforets) [inline src/Sets/Ball2.jl]:
```suggestion
    D = Vector{Vector{N}}(undef, nsamples) # preallocate output
```

**Comment by mforets** [comment]
Thx for the review!

**Pull request merged**